server/src/main/java/org/candlepin/model/AbstractHibernateObject.java

```diff
@@ -33,12 +33,17 @@
 
 /**
  * Abstract class for hibernate entities
+ *
+ * @param <T>
+ *  Entity type extending this class; should be the name of the subclass
  */
 @MappedSuperclass
 @XmlType(name = "CandlepinObject")
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonFilter("DefaultFilter")
-public abstract class AbstractHibernateObject implements Persisted, Serializable {
+public abstract class AbstractHibernateObject<T extends AbstractHibernateObject>
```

Comment:
It compiles, but it's not working as you expect. The ```public OwnerDTO populate(OwnerDTO source)``` is an overload of populate, so you end up with a ```populate(Object)``` (or maybe ```populate(CandlepinDTO)``` depending on how erasure works with scoped generics) and ```populate(OwnerDTO)```.

That's what I ran into while testing.